src/couch_replicator/src/couch_replicator_scheduler.erl

```diff
@@ -734,6 +775,24 @@ reset_job_process(#job{} = Job) ->
 
 -spec reschedule(#state{}) -> ok.
 reschedule(State) ->
+    % Charge all running jobs for the current interval
+    RunningJobs = running_jobs(),
+    Now = os:timestamp(),
+    lists:foreach(fun(Job) ->
+        couch_replicator_share:charge(Job, State#state.interval, Now)
+    end, RunningJobs),
+
+    % Update usage table
+    couch_replicator_share:update_usage(),
+
+    % Decay all the process priorities
+    couch_replicator_share:decay_priorities(),
+
+    % Adjust running process priorities
+    lists:foreach(fun(Job) ->
+        couch_replicator_share:update_priority(Job)
+    end, RunningJobs),
```

Comment:
This whole hunk feels as though it should be a single call to `couch_replicator_share` and that the multiple calls are leaking implementation internals outside of that module.

Reply:
Good idea. I was slightly worried about passing in the whole list of running jobs into the call